py/client/pydeephaven/ticking/table_listener.py

```diff
@@ -0,0 +1,117 @@
+                    rows: dhc.RowSequence,
+                    col_names: str | List[str] | None,
+                    chunk_size: int) -> DictGeneratorType:
+    col_names = tick_util.canonicalize_cols_param(table, col_names)
+    while not rows.empty:
+        these_rows = rows.take(chunk_size)
+        rows = rows.drop(chunk_size)
+
+        result: ColDictType = {}
+        for i in range(len(col_names)):
+            col_name = col_names[i]
+            col = table.get_column_by_name(col_name, True)
+            [data, bools] = col.get_chunk(these_rows, True)
+            result[col_name] = (data, bools)
+
+        yield result
+
+def _sole_item(generator):
+    try:
+        return next(generator)
+
+    except StopIteration:
+        return {}
+
+class TableUpdate:
+    update: dhc.TickingUpdate
+
+    def __init__(self, update: dhc.TickingUpdate):
+        self.update = update
+
+    def removed(self, cols: str | List[str] | None = None) -> ColDictType:
+        return _sole_item(self.removed_chunks(self.update.removed_rows.size, cols))
+
+    def removed_chunks(self, chunk_size: int, cols: str | List[str] | None = None) -> DictGeneratorType:
+        return _make_generator(self.update.before_removes, self.update.removed_rows, cols, chunk_size)
+
+    def added(self, cols: str | List[str] | None = None) -> ColDictType:
+        return _sole_item(self.added_chunks(self.update.added_rows.size, cols))
+
+    def added_chunks(self, chunk_size: int, cols: str | List[str] | None = None) -> DictGeneratorType:
+        return _make_generator(self.update.after_adds, self.update.added_rows, cols, chunk_size)
+
+    def modified_prev(self, cols: str | List[str] | None = None) -> ColDictType:
+        return _sole_item(self.modified_prev_chunks(self.update.all_modified_rows.size, cols))
+
+    def modified_prev_chunks(self, chunk_size: int, cols: str | List[str] | None = None) -> DictGeneratorType:
+        return _make_generator(self.update.before_modifies, self.update.all_modified_rows, cols, chunk_size)
+
+    def modified(self, cols: str | List[str] | None = None) -> ColDictType:
+        return _sole_item(self.modified_chunks(self.update.all_modified_rows.size, cols))
+
+    def modified_chunks(self, chunk_size: int, cols: str | List[str] | None = None) -> DictGeneratorType:
+        return _make_generator(self.update.after_modifies, self.update.all_modified_rows, cols,chunk_size)
+
+class TableListener(ABC):
+    @abstractmethod
+    def on_update(self, update: TableUpdate) -> None:
+        pass
+
+    @abstractmethod
+    def on_error(self, error: Exception):
+        pass
+
+
+class TableListenerHandle:
+    _table: dhc.Table
+    _listener: TableListener
+    _cancelled: bool
+    _bp: dhc.BarrageProcessor
+    _writer: flight.FlightStreamWriter
+    _reader: flight.FlightStreamReader
+    _thread: threading.Thread
+
+    def __init__(self, table: dhc.Table, listener: TableListener):
+        self._table = table
```

Comment:
should this be pydeephaven.table.Table?

Reply:
No. I think the issue is a terminology collision.

`pydeephaven.table.Table` is your wrapper that holds a session, ticket, etc and can be used to chain together server operations. On the other hand, `pydeephaven.ticking.Table` is a wrapper of the C++ object that allows access to the dynamically-updating local copy of the server table that has been subscribed to. It has very different methods, like `get_row_sequence` and `get_column` that allow for actually accessing the data in the columns of that local copy of the table.

Based on your comment, I think to avoid confusion, we should give this a very different name. I'm not sure exactly what though. `CachedTable`, `LocalTable`, `TickingTable`, `MirroredTable`, `TableSnapshot` ..... I'm not really sure. Do you have a suggestion?